### PR TITLE
#9113 fix url_path attribute value is not populated

### PR DESCRIPTION
**Pull request by thiagolima-bm**
Fix url_path attribute value is not populated for any product #9113

### CODE REVIEW

File: app/code/Magento/CatalogUrlRewrite/Observer/ProductUrlPathAutogeneratorObserver.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+use Magento\Catalog\Model\Product;
+use Magento\CatalogUrlRewrite\Model\ProductUrlPathGenerator;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+class ProductUrlPathAutogeneratorObserver implements ObserverInterface
+{
+    /** @var ProductUrlPathGenerator */
+    protected $productUrlPathGenerator;
```
Comment (ihor-sviziev) [inline app/code/Magento/CatalogUrlRewrite/Observer/ProductUrlPathAutogeneratorObserver.php]: According to Magento technical guidelines 
http://devdocs.magento.com/guides/v2.1/coding-standards/technical-guidelines/technical-guidelines.html
2.7. All non-public properties and methods SHOULD be private.

Could you change protected visibility to private?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by thiagolima-bm** [comment]
@ihor-sviziev I have removed the observer that I created. I would say that is better to use the same observer ProductUrlKeyAutogeneratorObserver and just add a new line instead
`$product->setUrlPath($this->productUrlPathGenerator->getUrlPath($product));`

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
@thiagolima-bm not sure that setting url_path should be added to existing class because of class name, but in general looks good.
now integration tests failed. Could you adjust them jf needed?

**Comment by thiagolima-bm** [comment]
@ihor-sviziev I will take a look why the tests are failing

File: app/code/Magento/CatalogUrlRewrite/Observer/ProductUrlKeyAutogeneratorObserver.php [hunk 2]
```diff
@@ -32,5 +32,6 @@ public function execute(\Magento\Framework\Event\Observer $observer)
         /** @var Product $product */
         $product = $observer->getEvent()->getProduct();
         $product->setUrlKey($this->productUrlPathGenerator->getUrlKey($product));
+        $product->setUrlPath($this->productUrlPathGenerator->getUrlPath($product));
```
Comment (TeknoFiend) [inline app/code/Magento/CatalogUrlRewrite/Observer/ProductUrlKeyAutogeneratorObserver.php]: The issue mentioned using `[product-suffix-from-setting]` to build the path.  Should this not be `getUrlPathWithSuffix`?

Sorry for the drive-by comment, I'm in need of a populated path, that includes the suffix.

**Comment by thiagolima-bm** [comment]
@ihor-sviziev any clue where this test could be failing?

File: app/code/Magento/CatalogUrlRewrite/Observer/ProductUrlPathAutoGeneratorObserver.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    /**
+     * @param \Magento\Framework\Event\Observer $observer
+     * @return void
+     */
+    public function execute(Observer $observer)
+    {
+        /** @var Product $product */
+        $product = $observer->getEvent()->getProduct();
+        $product->setUrlKey($this->productUrlPathGenerator->getUrlPath($product));
```
Comment (ihor-sviziev) [inline app/code/Magento/CatalogUrlRewrite/Observer/ProductUrlPathAutoGeneratorObserver.php:34]: 1. Looks like you was thinking to us $product->setUrlPath() method instead
2. Would be great to add some unit or integration test there. Could you add it?

If you need any help - feel free to ask

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/CatalogUrlRewrite/Observer/ProductUrlPathAutoGeneratorObserver.php (hunk 3 above)
Comment (thiagolima-bm) [reply app/code/Magento/CatalogUrlRewrite/Observer/ProductUrlPathAutoGeneratorObserver.php:34]: For sure! I need more coffee :P

**Comment by ihor-sviziev** [comment]
Hi @thiagolima-bm, are you interested in finishing your PR or maybe you need some help?

**Comment by okorshenko** [comment]
Hi @thiagolima-bm 
We are closing this PR due to inactivity. Feel free to reopen it once ready
